Sources/SwiftKafka/KafkaProducer.swift

```diff
@@ -14,21 +14,246 @@
 
 import Crdkafka
 import Logging
+import NIOCore
+
+/// `NIOAsyncSequenceProducerBackPressureStrategy` that always returns true.
+struct NoBackPressure: NIOAsyncSequenceProducerBackPressureStrategy {
+    func didYield(bufferDepth: Int) -> Bool { true }
+    func didConsume(bufferDepth: Int) -> Bool { true }
+}
+
+/// `NIOAsyncSequenceProducerBackPressureStrategy` that does nothing.
+struct NoDelegate: NIOAsyncSequenceProducerDelegate {
+    func produceMore() {}
+    func didTerminate() {}
+}
+
+/// `AsyncSequence` implementation for handling messages acknowledged by the Kafka cluster (``KafkaAcknowledgedMessage``).
+public struct AcknowledgedMessagesAsyncSequence: AsyncSequence {
+    public typealias Element = KafkaAcknowledgedMessage
+
+    let _internal = NIOAsyncSequenceProducer.makeSequence(
```

Comment:
We should not be holding the result of the `makeSequence` call here. This way you are retaining both the `NIOAsyncSequenceProducer` and the `NIOAsyncSequenceProducer.Source`. We should rather pass in the former into an init of this sequence and hold the `Source` inside the `KafaProducer`. This means calling `makeSequence` in the `init` of the producer.

Reply:
Oh, must have overlooked it in the documentation 😅